koog-ktor/src/commonMain/kotlin/ai/koog/ktor/utils/EnvConfigLoader.kt

```diff
@@ -0,0 +1,146 @@
+package ai.koog.ktor.utils
+
+import ai.koog.ktor.KoogAgentsConfig
+import ai.koog.prompt.llm.LLMProvider
+import io.ktor.server.application.ApplicationEnvironment
+import io.ktor.server.config.ApplicationConfig
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * Loads and configures the environment-specific settings for Koog agents based on the provided
+ * application configuration. This includes setup for OpenAI, Anthropic, Google, OpenRouter,
+ * Ollama, as well as default and fallback LLM (Large Language Model) configurations.
+ *
+ * @param envConfig The application configuration that contains environment-specific properties
+ *                  for configuring Koog agents and associated integrations.
+ * @return A populated instance of [KoogAgentsConfig] with the environment-specific settings applied.
+ */
+internal fun ApplicationEnvironment.loadAgentsConfig(): KoogAgentsConfig {
+    val koogConfig = KoogAgentsConfig()
+        .openAI(config)
+        .anthropic(config)
+        .google(config)
+        .openrouter(config)
+
+    if (config.propertyOrNull("koog.ollama.enable") != null) {
```

Comment:
Why is this check for ollama?
Then why isn’t it there for the other providers?

Reply:
Other providers are enabled by the presence of an `apiKey`, but since `ollama` doesn't work with `apiKeys` it should be explicitly enabled.

Other alternative is to _attempt_ connecting to Ollama, and always enabling it but that seemed dangerous.